src/panels/config/logs/error-log-card.ts

```diff
@@ -1,102 +1,185 @@
+  },
+];
+
+@customElement("error-log-card")
 class ErrorLogCard extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
   @property() public filter = "";
 
   @state() private _isLogLoaded = false;
 
-  @state() private _errorHTML!: TemplateResult[] | string;
+  @state() private _logHTML!: TemplateResult[] | string;
+
+  @state() private _error?: string;
+
+  @state() private _selectedLogProvider = "core";
+
+  @state() private _hassioLoaded = false;
 
   protected render(): TemplateResult {
     return html`
       <div class="error-log-intro">
-        ${this._errorHTML
+        ${this._error
+          ? html`<ha-alert alert-type="error">${this._error}</ha-alert>`
+          : ""}
+        ${this._logHTML
           ? html`
               <ha-card>
-                <ha-icon-button
-                  .path=${mdiRefresh}
-                  @click=${this._refreshErrorLog}
-                  .label=${this.hass.localize("ui.common.refresh")}
-                ></ha-icon-button>
-                <div class="card-content error-log">${this._errorHTML}</div>
+                <div class="header">
+                  ${this.hass.userData?.showAdvanced && this._hassioLoaded
+                    ? html`
+                        <ha-select
+                          .label=${this.hass.localize(
+                            "ui.panel.config.logs.log_provider"
+                          )}
+                          @selected=${this._setLogProvider}
+                          .value=${this._selectedLogProvider}
+                        >
+                          ${logProviders.map(
+                            (provider) => html`
+                              <mwc-list-item .value=${provider.key}>
+                                ${provider.name}
+                              </mwc-list-item>
+                            `
+                          )}
+                        </ha-select>
+                      `
+                    : ""}
+                  <ha-icon-button
+                    .path=${mdiRefresh}
+                    @click=${this._refresh}
+                    .label=${this.hass.localize("ui.common.refresh")}
+                  ></ha-icon-button>
+                </div>
+                <div class="card-content error-log">${this._logHTML}</div>
               </ha-card>
             `
-          : html`
-              <mwc-button raised @click=${this._refreshErrorLog}>
+          : ""}
+        ${!this._logHTML
+          ? html`
+              <mwc-button raised @click=${this._refreshLogs}>
                 ${this.hass.localize("ui.panel.config.logs.load_full_log")}
               </mwc-button>
-            `}
+            `
+          : ""}
       </div>
     `;
   }
 
-  protected firstUpdated(changedProps) {
+  protected firstUpdated(changedProps: PropertyValues) {
     super.firstUpdated(changedProps);
 
+    if (isComponentLoaded(this.hass, "hassio")) {
+      this._hassioLoaded = true;
+    }
```

Comment:
Not sure if `hassio` is always already loaded when this page is first updated, would probably keep this in `render` and not in firstupdated